src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/OverloadMethodsDeclarationOrderCheckTest.java

```diff
@@ -75,4 +183,38 @@ public void testTokensNotNull() {
             .isNotNull();
     }
 
+    @Test
+    public void testOverloadDescriptorEquals()
```

Comment:
Pure unit test is not much better than suppression, we avoid this technical debt at all costs. Please remove this test so that we can see/ work through all pitest failures. If code is not being hit, remove it and run check regression report to help find cases to cover new code.